lib/base.sh

```diff
@@ -10,6 +10,37 @@ lib="${ch_bin}/../lib/charliecloud"
 . "${lib}/version.sh"
 
 
+architecture_choose () {
+    host_arch=$(uname -m)
+    case $host_arch in
+        aarch32)
+            arch='arm/v7'
+            ;;
+        aarch64)
+            arch='arm64/v8'
+            ;;
+        armv5l)
+            arch='arm/v5'
+            ;;
+        armv6l)
+            arch='arm/v6'
+            ;;
+        armv7l)
+            arch='arm/v7'
+            ;;
+        armv8l)
+            arch='arm64/v8'
+            ;;
+        x86_64)
+            arch='amd64'
+            ;;
+        *)
+            arch="$host_arch"
+            ;;
+    esac
+    export CH_GROW_ARCH=$arch
+}
```

Comment:
1. This is a lot of lines to implement a table. Is there anything shorter that we can do?
2. I wonder if we should use dot instead of slash or percent to separate architecture and variant internally?
3. We should separate the concerns of (1) mapping arch specifications to the canonical one and (2) setting what arch to use.

Reply:
1. If we wish to remain explicit we could condense it to:
```bash
    host_arch=$(uname -m)
    case $host_arch in
        aarch32) arch='arm/v7'     ;;
        aarch64) arch='arm64/v8'   ;;
         armv5l) arch='arm/v5'     ;;
         armv6l) arch='arm/v6'     ;;
         armv7l) arch='arm/v7'     ;;
         armv8l) arch='arm64/v8'   ;;
         x86_64) arch='amd64'      ;;
              *) arch="$host_arch" ;;
    esac
```
2. OK

3. we could always store at `$CH_GROW_STORAGE/$(uname -m) and let ch-image worry about converting and inferring canonical name.